Kernel/Process.cpp

```diff
@@ -3263,9 +3263,14 @@ int Process::sys$getsockname(int sockfd, sockaddr* addr, socklen_t* addrlen)
 
     auto& socket = *description->socket();
     REQUIRE_PROMISE_FOR_SOCKET_DOMAIN(socket.domain());
-    if (!socket.get_local_address(addr, addrlen))
+
+    OwnPtr<sockaddr> addr((sockaddr*)operator new(addrlen));
+
+    if (!socket.get_local_address(addr, &addrlen))
         return -EINVAL; // FIXME: Should this be another error? I'm not sure.
 
+    copy_to_user(user_addr, addr, addrlen);
+    copy_to_user(user_addrlen, &addrlen, sizeof(*user_addrlen));
```

Comment:
This can use the templated version, no?

```suggestion
    copy_to_user(user_addrlen, &addrlen);
```